sessiontxn/staleread/provider.go

```diff
@@ -60,43 +63,81 @@ func (p *StalenessTxnContextProvider) GetStmtForUpdateTS() (uint64, error) {
 func (p *StalenessTxnContextProvider) OnInitialize(ctx context.Context, tp sessiontxn.EnterNewTxnType) error {
 	switch tp {
 	case sessiontxn.EnterNewTxnDefault, sessiontxn.EnterNewTxnWithBeginStmt:
-		if err := p.sctx.NewStaleTxnWithStartTS(ctx, p.ts); err != nil {
-			return err
-		}
-		p.is = p.sctx.GetSessionVars().TxnCtx.InfoSchema.(infoschema.InfoSchema)
-		if err := p.sctx.GetSessionVars().SetSystemVar(variable.TiDBSnapshot, ""); err != nil {
-			return err
-		}
+		return p.enterNewStaleTxn(ctx)
 	case sessiontxn.EnterNewTxnWithReplaceProvider:
-		if p.is == nil {
-			is, err := GetSessionSnapshotInfoSchema(p.sctx, p.ts)
-			if err != nil {
-				return err
-			}
-			p.is = temptable.AttachLocalTemporaryTableInfoSchema(p.sctx, is)
-		}
+		return p.enterNewStaleTxnWithReplaceProvider(ctx)
 	default:
 		return errors.Errorf("Unsupported type: %v", tp)
 	}
+}
+
+func (p *StalenessTxnContextProvider) enterNewStaleTxn(ctx context.Context) error {
```

Comment:
I think it's name can be `activateTxn` or `activateStaleTxn` or just merge the codes to `ActivateTxn `

Reply:
Done.